### PR TITLE
[SLS-1361] Use DD_XXX env vars for tagging

### PR DESCRIPTION
**Pull request by lizapressman**
### What does this PR do?

When using the extension (extensionVersion is set), we now add `DD_ENV`, `DD_SERVICE`, `DD_VERSION`, and `DD_TAGS` as environment variables if we have set env, service, version, and tags like so: 
Screen Shot 2022-04-19 at 2 22 02 PM

If the forwarderArn is provided, we also add these as tags (service, env, version, and parsed tags).

### Motivation

https://datadoghq.atlassian.net/browse/SLS-1361

### Testing Guidelines

Added these to the datadogCDK configuration in the `index.ts` files for both v1 and v2 and verified that the outcome was what I expected after running `diff`

### Types of Changes

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change 
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [x] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [x] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

File: v1/src/datadog.ts [hunk 1]
```diff
@@ -112,3 +115,29 @@ export function addCdkConstructVersionTag(lambdaFunctions: lambda.Function[]) {
     });
   });
 }
+
+function setTags(lambdaFunctions: lambda.Function[], props: DatadogProps) {
```
Comment (hghotra) [inline v1/src/datadog.ts:119]: Can we add a unit test for this?

**Review by hghotra — APPROVED** [review]
👍🏼

File: README.md [hunk 2]
```diff
@@ -135,7 +139,12 @@ _Note_: The descriptions use the npm package parameters, but they also apply to
...
+| `service` | `service` | When set along with `extensionLayerVersion`, a `DD_SERVICE` environment variable is added to all Lambda functions with the provided value. When set along with `forwarderArn`, a `service` tag is added to all Lambda functions with the provided value. |
+| `version` | `version` | When set along with `extensionLayerVersion`, a `DD_VERSION` environment variable is added to all Lambda functions with the provided value. When set along with `forwarderArn`, a `version` tag is added to all Lambda functions with the provided value. |
+| `tags` | `tags` | A comma separated list of key:value pairs as a single string. When set along with `extensionLayerVersion`, a `DD_TAGS` environment variable is added to all Lambda functions with the provided value. When set along with `forwarderArn`, the cdk parses the string and sets each key:value pair as a tag to all Lambda functions. |
 
+Note: env, service, version, and tags will override function level DD_XXX environment variables.
```
Comment (cswatt) [inline README.md:147]:
```suggestion
**Note**: `env`, `service`, `version`, and `tags` override function level `DD_XXX` environment variables.
```

File: README.md [hunk 3]
```diff
@@ -216,7 +225,7 @@ Add tags to your constructs. We recommend setting an `env` and `service` tag to
 
 The Datadog CDK construct takes in a list of lambda functions and installs the Datadog Lambda Library by attaching the Lambda Layers for [Node.js][2] and [Python][1] to your functions. It redirects to a replacement handler that initializes the Lambda Library without any required code changes. Additional configurations added to the Datadog CDK construct will also translate into their respective environment variables under each lambda function (if applicable / required).
 
-While Lambda function based log groups are handled by the `addLambdaFunctions` method automatically, the construct has an additional function `addForwarderToNonLambdaLogGroups` which will subscribe the forwarder to any additional log groups of your choosing. 
+While Lambda function based log groups are handled by the `addLambdaFunctions` method automatically, the construct has an additional function `addForwarderToNonLambdaLogGroups` which will subscribe the forwarder to any additional log groups of your choosing.
```
Comment (cswatt) [inline README.md:228]:
```suggestion
While Lambda function based log groups are handled by the `addLambdaFunctions` method automatically, the construct has an additional function `addForwarderToNonLambdaLogGroups` which subscribes the forwarder to any additional log groups of your choosing.
```

**Pull request merged**